src/psyclone/dynamo0p3.py

```diff
@@ -2019,7 +2019,13 @@ def initialise(self, parent):
         :raises InternalError: if an unsupported mesh property is encountered.
 
         '''
-        if not self._properties:
+        need_colour_limits = False
+        for call in self._calls:
+            if call.is_coloured() and not call.is_intergrid:
+                need_colour_limits = True
+                break
+
+        if not self._properties and not need_colour_limits:
```

Comment:
Comments added.

I could extend e.g. `Dynamo0p3ColourTrans.apply()` to add the necessary property to the Collection within the DynInvoke. However, that feels like it could be a bit fragile (e.g. if a user manipulates the tree in an unexpected way) and I'd rather leave it here and this way we'll always get it right. However, am happy to discuss further.

Reply:
No, I'm happy with your answer, I just wondered.